### PR TITLE
Feature/validation json

### PR DESCRIPTION
**Pull request by tdule-fairway**
Reapplied support for validations in json format (see original commit 4848a58 on 9/5/2017). Supports entire content package, single item package, or zip file.

### CODE REVIEW

File: TabulateSmarterTestContentPackage/ItemBankPackage.cs [hunk 1]
```diff
@@ -12,18 +12,32 @@ namespace TabulateSmarterTestContentPackage
 {
     class ItemBankPackage : TestPackage
     {
+        bool m_localPackage;
```
[the remark this answers is not in the document]
Comment (tdule-fairway) [reply TabulateSmarterTestContentPackage/ItemBankPackage.cs]: Nope not C#, more likely to see in C, Java and VB6, but it was more about blending with the existing code style :)

**Comment by jtreuting** [comment]
This looks good to me.  Once we are good with it, we will bring in Smarter Balanced and ask them to review and merge.  We should not do the merge ourselves.

**Comment by bredd** [comment]
I have three concerns with the implementation (not with the features):
 * Single Item Packages are implemented as a mode of ItemBankPackage. Instead, it should be a distinct implementation of ItemPackage.
 * It uses the NewtonSoft JSON component. We prefer to use System.Runtime.Serialization.Json thereby avoiding third-party dependencies. (Third-party dependencies have caused problems with producing a self-contained distribution.)
 * Most of the conditional code related to single-item tabulation should not be necessary. The comments suggest this is necessary to  make sure that tutorials and passages are tabulated even if there's not an item that depends on them. However, the dependency-related code is only a factor if you use the -ids command-line parameter; otherwise everything in the package is included.

Here's our plan:
 1. Smarter Balanced will accept the pull request, then run the application in single-item mode and with JSON output to establish a functional baseline.
 2. I will create a feature/merge branch in which I will roll-back the unneded conditional code and address the other issues above.
 3. Smarter Balanced will test the feature/merge branch to ensure it produces the same output as in step 1.
 4. We will use a pull request to merge back into the feature branch.
 5. Fairway tests the feature branch in-context to ensure that it works as expected
 6. Smarter Balanced merges into the master branch and produces a release.

**Comment by tdule-fairway** [comment]
@bredd With regards to this comment:
> Most of the conditional code related to single-item tabulation should not be necessary. 

There are two scenarios that required the conditional code - 1) processing a single item with dependency on a tutorial or wordlist and 2) processing a single item that _is_ a tutorial or wordlist.

For the first, I found that while processing a single item, if a dependency on a wordlist or tutorial is found in the item xml, it gets added to the respective queue, and then later on tries to process them. This causes an issue since the dependency items are not also part of the package being processed. So the desired outcome is to skip those checks, rather than recursively add them to the package as they are found.

For the second scenario, this seemed to not be supported (perhaps deliberately), and rather than create a potential bug, the condition ensures that only single item processing will allow for standalone tutorial or wordlist package processing. All other previous valid input will continue to work as expected.

I hope this explanation is clear. Please reach out if I can provide additional clarification.

**Pull request merged**